include/ed/loop_usage_status.h

```diff
@@ -0,0 +1,169 @@
+
+#include <diagnostic_msgs/DiagnosticStatus.h>
+
+#include <tue/profiling/loop_timer.h>
+#include <tue/profiling/timer.h>
+
+#include <math.h>
+
+
+namespace ed
+{
+
+/**
+   * \brief A diagnostic task that monitors the frequency of an event.
+   *
+   * This diagnostic task monitors the frequency and usage of a loop and creates corresponding diagnostics.
+   * It will report a warning if the frequency is outside acceptable bounds, and report an error
+   * if there have been no events in the latest window.
+   * Heavily inspired by diagnostic_updater::FrequencyStatus
+   */
+
+class LoopUsageStatus : public diagnostic_updater::DiagnosticTask
+{
+private:
+    const diagnostic_updater::FrequencyStatusParam params_;
+
+    tue::LoopTimer timer_;
+    std::vector<long double> starts_; // Micro-seconds
+    std::vector<long double> durations_; // Seconds
+    std::vector<int> seq_nums_;
+    int hist_indx_;
+    long double start_;
+    long double duration_;
+
+    boost::mutex lock_;
+
+public:
+    /**
+    * \brief Constructs a LoopUsageStatus class with the given parameters.
+    */
+    LoopUsageStatus(const diagnostic_updater::FrequencyStatusParam &params, std::string name) :
+    DiagnosticTask(name), params_(params),
+    starts_(params_.window_size_), durations_(params_.window_size_), seq_nums_(params_.window_size_)
+    {
+        clear();
+    }
+
+    /**
+    * \brief Constructs a LoopUsageStatus class with the given parameters.
+    *        Uses a default diagnostic task name of "Loop Usage Status".
+    */
+
+    LoopUsageStatus(const diagnostic_updater::FrequencyStatusParam &params) :
+    DiagnosticTask("Loop Usage Status"), params_(params),
+    starts_(params_.window_size_), durations_(params_.window_size_), seq_nums_(params_.window_size_)
+    {
+        clear();
+    }
+
+    /**
+     * @brief Expose const reference to loop timer. No call to the const timer will be able to influence the behaviour of this class.
+     * @return const reference to the loop timer
+     */
+    const tue::LoopTimer& getTimer() const { return timer_; }
+
+    /**
+    * \brief Resets the statistics.
+    */
+    void clear()
+    {
+        boost::mutex::scoped_lock lock(lock_);
+        timer_.reset();
+        long double curtime = tue::Timer::nowMicroSec();
+
+        std::fill(starts_.begin(), starts_.end(), curtime);
+        std::fill(durations_.begin(), durations_.end(), 0);
+        std::fill(seq_nums_.begin(), seq_nums_.end(), timer_.getIterationCount());
+
+        hist_indx_ = 0;
+    }
+
+    /**
+    * \brief Signals that a start event has occurred.
+    */
+    void start()
+    {
+        boost::mutex::scoped_lock lock(lock_);
+        timer_.start();
+    }
+
+    /**
+    * \brief Signals that a stop event has occurred.
+    */
+    void stop()
+    {
+        boost::mutex::scoped_lock lock(lock_);
+        timer_.stop();
+    }
+
+    virtual void run(diagnostic_updater::DiagnosticStatusWrapper &stat)
```

Comment:
Add docstring, describe the desired behaviour. and what a user may expect from this method

Reply:
This is inherited from the baseclass...

Reply:
Maar dat was niet...

I just added a simple docstring. (Aka copied it from `diagnostic_updater::FrequencyStatus`)